components/automate-cli/cmd/chef-automate/config.go

```diff
@@ -169,6 +187,70 @@ var setConfigCmd = &cobra.Command{
 	},
 }
 
+func runGenCmd(cmd *cobra.Command, args []string) error {
+	fsu := fileutils.NewFileSystemUtils()
+	p := pmt.PromptFactory(os.Stdin, os.Stdout, fsu)
+	g := genconfig.GenConfigImpFactory(p)
+
+	err := g.GenConfigWithPrompts()
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FailedToGenConfig,
+			"Failed to generate config with given input values to the prompts.",
+		)
+	}
+	t, err := g.Toml()
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.MarshalError,
+			"Marshaling configuration to TOML failed",
+		)
+	}
+
+	// Handle writing to a file if a path was given
+	if len(args) > 0 && args[0] != "" {
+		outFile, err := filepath.Abs(args[0])
+		if err != nil {
+			return status.Annotate(err, status.FileAccessError)
+		}
+
+		err = checkConfigGenFileExist(outFile, fsu)
+		if err != nil {
+			return err
+		}
+		err = os.WriteFile(outFile, t, 0644)
+		if err != nil {
+			return status.Wrap(err, status.FileAccessError, fmt.Sprint("Failed to write to file: ", outFile))
+		}
+	} else {
+		writer.Println("")
+		writer.Println(string(t))
+	}
+
+	status.GlobalResult = t
+	return nil
+}
+
+func checkConfigGenFileExist(outFile string, fsu fileutils.FileUtils) error {
+	if _, err := os.Stat(outFile); err == nil {
```

Comment:
Use `github.com/chef/automate/lib/io/fileutils` package instead of directly using the native package.

Reply:
fixed